auslib/test/blobs/test_systemaddons.py

```diff
@@ -222,3 +222,165 @@ def testDoesNotContainForbiddenDomain(self):
+                    "p": {
+                        "alias": "q"
+                    },
+                    "q": {
+                        "filesize": 2,
+                        "hashValue": "3",
+                        "fileUrl": "http://boring.com/blah"
+                    }
+                }
+            }
+        }
+    }
+    """)
+        blob.validate('gg', self.whitelistedDomains)
+
+        # Test5: Incorrect layout with missing version for an addon name
+
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+        "name": "fake",
+        "schema_version": 5000,
+        "hashFunction": "SHA512",
+        "addons": {
+            "c": {
+                "platforms": {
+                    "p": {
+                        "alias": "q"
+                    },
+                    "q": {
+                        "filesize": 2,
+                        "hashValue": "3",
+                        "fileUrl": "http://boring.com/blah"
+                    }
+                }
+            }
+        }
+    }
+    """)
+        self.assertRaises(Exception, blob.validate, 'gg', self.whitelistedDomains)
+
+        # Test6: Correct layout with empty platforms
+
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+        "name": "fake",
+        "schema_version": 5000,
+        "hashFunction": "SHA512",
+        "addons": {
+            "c": {
+                "version": "1",
+                "platforms": {}
+            }
+        }
+    }
+    """)
+        blob.validate('gg', self.whitelistedDomains)
+
+        # Test7: Incorrect layout with empty platform name
+
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+        "name": "fake",
+        "schema_version": 5000,
+        "hashFunction": "SHA512",
+        "addons": {
+            "c": {
+                "version": "1",
+                "platforms": {
+                    "p": {},
+                    "q": {
+                        "filesize": 2,
+                        "hashValue": "3",
+                        "fileUrl": "http://boring.com/blah"
+                    }
+                }
+            }
+        }
+    }
+    """)
+        self.assertRaises(Exception, blob.validate, 'gg', self.whitelistedDomains)
+
+        # Test8: Incorrect layout with missing filesize
+
+        blob = SystemAddonsBlob()
+        blob.loadJSON("""
+    {
+        "name": "fake",
+        "schema_version": 5000,
+        "hashFunction": "SHA512",
+        "addons": {
+            "c": {
+                "version": "1",
+                "platforms": {
+                    "p": {
+                        "alias": "q"
+                    },
+                    "q": {
+                        "hashValue": "3",
+                        "fileUrl": "http://boring.com/blah"
+                    }
+                }
+            }
+        }
+    }
+    """)
+        self.assertRaises(Exception, blob.validate, 'gg', self.whitelistedDomains)
```

Comment:
You've done a great job writing all of these tests, but because they are all testing different cases, they should each be put into their own test. The way things are right now, if the first test fails the rest will not run, so it's more difficult to get an overview of failures. Putting them into their own methods will ensure that each runs no matter what happens with the others.